### ISSUE TITLE
[1.2.0] Segmentation Violation

### ISSUE THREAD

**Issue by rossmcdonald** [influxdata/influxdb#7922]
```
version 1.2.0, branch master, commit b7bb7e8359642b6e071735b50ae41f5eb343fd42
```

Encountered this error when backfilling:

```
fatal error: fault
[signal SIGSEGV: segmentation violation code=0x1 addr=0x7fd7650c2fb5 pc=0x45fc8a]

goroutine 1654 [running]:
runtime.throw(0xab4597, 0x5)
        /usr/local/go/src/runtime/panic.go:566 +0x95 fp=0xc421d4d538 sp=0xc421d4d518
runtime.sigpanic()
        /usr/local/go/src/runtime/sigpanic_unix.go:27 +0x288 fp=0xc421d4d590 sp=0xc421d4d538
runtime.memeqbody()
        /usr/local/go/src/runtime/asm_amd64.s:1423 +0x8a fp=0xc421d4d598 sp=0xc421d4d590
runtime.mapaccess2_faststr(0xa5b580, 0xc4b0d72120, 0x7fd7650c2fb5, 0xd, 0xd, 0x0)
        /usr/local/go/src/runtime/hashmap_fast.go:323 +0x104 fp=0xc421d4d5f8 sp=0xc421d4d598
github.com/influxdata/influxdb/tsdb.(*Store).TagValues(0xc42000b600, 0xc46400e028, 0xa, 0xe2eba0, 0xc4bfb45860, 0x6, 0xc4363d2a58, 0x4ed962, 0xe4abc0, 0xc45f57d4a0)
        /root/go/src/github.com/influxdata/influxdb/tsdb/store.go:932 +0x693 fp=0xc421d4d9a8 sp=0xc421d4d5f8
github.com/influxdata/influxdb/coordinator.LocalTSDBStore.TagValues(0xc42000b600, 0xc46400e028, 0xa, 0xe2eba0, 0xc4bfb45860, 0xc45f57d4a0, 0xc45f57d4a0, 0x7fd8b1caa000, 0xc4363d2b28, 0x4b8a78)
        <autogenerated>:124 +0x70 fp=0xc421d4da08 sp=0xc421d4d9a8
github.com/influxdata/influxdb/coordinator.(*StatementExecutor).executeShowTagValues(0xc4201b64d0, 0xc45b61c800, 0xc421d4dc78, 0x7fd8b1d32178, 0xc4201d0160)
        /root/go/src/github.com/influxdata/influxdb/coordinator/statement_executor.go:852 +0x93 fp=0xc421d4db70 sp=0xc421d4da08
github.com/influxdata/influxdb/coordinator.(*StatementExecutor).ExecuteStatement(0xc4201b64d0, 0xe328a0, 0xc45b61c800, 0x0, 0x2f, 0xc4c20eb140, 0xc47bbbc720, 0xe35b60, 0xc4201bf1a0, 0xc4c20eb080, ...)
        /root/go/src/github.com/influxdata/influxdb/coordinator/statement_executor.go:177 +0x2a9b fp=0xc421d4dc60 sp=0xc421d4db70
github.com/influxdata/influxdb/influxql.(*QueryExecutor).executeQuery(0xc42019eae0, 0xc4639c4e00, 0xc46400e028, 0xa, 0x2710, 0x1, 0x0, 0x0, 0xc47bbbc660, 0xc47bbbc600, ...)
        /root/go/src/github.com/influxdata/influxdb/influxql/query_executor.go:312 +0x55d fp=0xc421d4df48 sp=0xc421d4dc60
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2086 +0x1 fp=0xc421d4df50 sp=0xc421d4df48
created by github.com/influxdata/influxdb/influxql.(*QueryExecutor).ExecuteQuery
        /root/go/src/github.com/influxdata/influxdb/influxql/query_executor.go:207 +0xb5
```

Full stack trace: [trace.tar.gz](https://github.com/influxdata/influxdb/files/743008/trace.tar.gz)

**Comment by jwilder** [comment]
I think the issue might be that https://github.com/influxdata/influxdb/blob/1.2/tsdb/store.go#L932 is accessing `Series.Tags` without an `RLock`.  When a TSM file is closed, if the tags are from that file they are dereferenced which creates a copy of them under a `Lock`.  See: https://github.com/influxdata/influxdb/blob/master/tsdb/meta.go#L1499  This panic might have occurred because the series/tags were we accessed while some of the series were being dereferenced.

@benbjohnson What do you think?

**Comment by benbjohnson** [comment]
@jwilder Yes, it looks like a dereferencing issue. I added a fix here: https://github.com/influxdata/influxdb/pull/7929

**Comment by jwilder** [comment]
Fixed via #7929
